pkg/lib/template/template_loader.go

```diff
@@ -282,6 +283,42 @@ func (loader *Loader) Render(
 		"match": func(regex, text string) (bool, error) {
 			return regexp.Match(regex, []byte(text))
 		},
+		"replace_http_server": func(srcURL string) (string, error) {
+			// If no override provided, return original one
+			if loader.HTTPServer == "" {
+				return srcURL, nil
+			}
+			// Parse source URL or fail
+			parsedURL, err := url.Parse(srcURL)
+			if err != nil {
+				return "", err
+			}
+			// Parse override URL or fail
+			parsedOverrideURL, err := url.Parse(loader.HTTPServer)
+			if err != nil {
+				return "", err
+			}
```

Comment:
Put this parsing into the very topp. Decorate your errors like

```
errors.Wrap(err, "replace_http_server failed")
```

Otherwise your caller does not know whats happening